crates/network/src/transaction/signer.rs

```diff
@@ -1,25 +1,52 @@
 use crate::{Network, TransactionBuilder};
 use alloy_consensus::SignableTransaction;
+use alloy_primitives::Address;
 use async_trait::async_trait;
+use futures_utils_wasm::impl_future;
 
 /// A signer capable of signing any transaction for the given network.
 ///
-/// Network crate authors should implement this trait on a type capable of signing any transaction
-/// (regardless of signature type) on a given network. Signer crate authors should instead implement
-/// [`TxSigner`] to signify signing capability for specific signature types.
+/// Network crate authors should implement this trait on a type capable of
+/// signing any transaction (regardless of signature type) on a given network.
+/// Signer crate authors should instead implement [`TxSigner`] to signify
+/// signing capability for specific signature types.
+///
+/// Network signers are expected to contain one or more signing credentials,
+/// keyed by signing address. The default signer address should be used when
+/// no specific signer address is specified.
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 pub trait NetworkSigner<N: Network>: std::fmt::Debug + Send + Sync {
+    /// Get the default signer address. This address should be used
+    /// in [`NetworkSigner::sign_transaction_from`] when no specific signer is
+    /// specified.
+    fn default_signer(&self) -> Address;
+
+    /// Asynchronously sign an unsigned transaction, with a specified
+    /// credential.
+    async fn sign_transaction_from(
+        &self,
+        sender: Option<Address>,
+        tx: N::UnsignedTx,
+    ) -> alloy_signer::Result<N::TxEnvelope>;
```

Comment:
this makes sense to me and is convenient to have,
it should be trivial for any, non-multiplex signers to implement by ensuring the default signer is Some(sender)

Reply:
doing a slight cleanup here as i wrote this function signature before i added the default signer

Reply:
removed the `Option` because we can rely on the caller to call `default_signer`